django/db/models/query.py

```diff
@@ -356,14 +357,35 @@ def __or__(self, other):
     ####################################
 
     def _iterator(self, use_chunked_fetch, chunk_size):
-        yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
+        iterable = self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size or 2000)
+        if not self._prefetch_related_lookups or chunk_size is None:
+            yield from iterable
+            return
+
+        iterator = iter(iterable)
+        while True:
+            results = list(islice(iterator, chunk_size))
+            if not results:
+                break
+            prefetch_related_objects(results, *self._prefetch_related_lookups)
+            yield from iter(results)
```

Comment:
Why is the `iter()` necessary here? Above we're calling `yield from iterable` so surely if `results` is an iterable - it is of type `list` - and not an iterator that will be fine?

Reply:
Yep, looks right to me. All the model instances are materialized as they have to be to do the prefetching, so just yield 'em on out!